source/Engine.cpp

```diff
@@ -1430,10 +1436,23 @@ void Engine::CalculateStep()
 
 	// Keep track of the flagship to see if it jumps or enters a wormhole this turn.
 	const Ship *flagship = player.Flagship();
+	bool flagshipWasCloaked = (flagship && flagship->Cloaking() != 0);
```

Comment:
Currently you're considering entering cloak as being fully cloaked. It might make sense to do it the other way around, i.e. you are only cloaked if you are fully cloaked, everything else is "uncloaked". Thoughts?

Reply:
Hm. It wasn't particularly intentional on my part, but it kind of feels like it makes sense? As soon as you start cloaking, your communications are cut off, and they aren't restored until you're fully uncloaked? And "encountering" an NPC seems like it would be done through the ship's comms...

Reply:
The game seems to already use the "you are only cloaked when you're fully cloaked" logic, for example when an enemy is attacking you and you cloak. The enemy ship continues to target and fire at you until you're fully cloaked.

Reply:
Ah, gotcha. I can definitely change it, then.

Reply:
OK; I think that should be all the spots where it was treating any cloak as all cloak switched.